packages/flutter/lib/src/rendering/table_border.dart

```diff
@@ -194,49 +190,49 @@ class TableBorder {
     assert(canvas != null);
     assert(rect != null);
     assert(rows != null);
-    assert(rows.length >= 2);
-    assert(rows.first == 0.0);
-    assert(rows.last == rect.height);
+    assert(rows.isEmpty || (rows.first > 0.0 && rows.last < rect.height));
     assert(columns != null);
-    assert(columns.isNotEmpty);
-    assert(columns.first == 0.0);
-    assert(columns.last < rect.width);
+    assert(columns.isEmpty || (columns.first > 0.0 && columns.last < rect.width));
 
     final Paint paint = new Paint();
     final Path path = new Path();
 
-    switch (verticalInside.style) {
-      case BorderStyle.solid:
-        paint
-          ..color = verticalInside.color
-          ..strokeWidth = verticalInside.width
-          ..style = PaintingStyle.stroke;
-        path.reset();
-        for (int x = 1; x < columns.length; x += 1) {
-          path.moveTo(rect.left + columns[x], rect.top);
-          path.lineTo(rect.left + columns[x], rect.bottom);
-        }
-        canvas.drawPath(path, paint);
-        break;
-      case BorderStyle.none:
-        break;
+    if (columns.isNotEmpty) {
+      switch (verticalInside.style) {
+        case BorderStyle.solid:
+          paint
+            ..color = verticalInside.color
+            ..strokeWidth = verticalInside.width
+            ..style = PaintingStyle.stroke;
+          path.reset();
+          for (double x in columns) {
+            path.moveTo(rect.left + x, rect.top);
+            path.lineTo(rect.left + x, rect.bottom);
+          }
+          canvas.drawPath(path, paint);
+          break;
+        case BorderStyle.none:
+          break;
+      }
     }
 
-    switch (horizontalInside.style) {
-      case BorderStyle.solid:
-        paint
-          ..color = horizontalInside.color
-          ..strokeWidth = horizontalInside.width
-          ..style = PaintingStyle.stroke;
-        path.reset();
-        for (int y = 1; y < rows.length; y += 1) {
-          path.moveTo(rect.left, rect.top + rows[y]);
-          path.lineTo(rect.right, rect.top + rows[y]);
-        }
-        canvas.drawPath(path, paint);
-        break;
-      case BorderStyle.none:
-        break;
+    if (rows.isNotEmpty) {
```

Comment:
and here you run the logic of the getRange iterator to check if the _rowTops list is empty, then you run it again for the for loop below.